pkg/utils/retry.go

```diff
@@ -0,0 +1,28 @@
+// Copyright 2020 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package utils
+
+// Retry when the when func returns true
+func Retry(f func() error, when func(error) bool) error {
+	e := f()
+	if e == nil {
+		return nil
+	}
```

Comment:
use for loop instead of Infinite recursion

Reply:
It's ok to use recursion here because it's a tail-recursive call.
Use loop will make this more complex.